packages/data/src/redux-store/metadata/selectors.js

```diff
@@ -54,6 +71,36 @@ export function hasFinishedResolution( state, selectorName, args = [] ) {
 	return getIsResolving( state, selectorName, args ) === false;
 }
 
+/**
+ * Returns true if resolution has failed for a given selector
+ * name, and arguments set.
+ *
+ * @param {State}     state        Data state.
+ * @param {string}    selectorName Selector name.
+ * @param {unknown[]} [args]       Arguments passed to selector.
+ *
+ * @return {boolean} Has resolution failed
+ */
+export function hasResolutionFailed( state, selectorName, args = [] ) {
+	const resolutionState = getResolutionState( state, selectorName, args );
+	const hasFailed = resolutionState && 'error' in resolutionState;
+	return !! hasFailed;
```

Comment:
You could as well return `!! getResolutionFailure( state, selectorName, args )`, it's equivalent.

Reply:
It's not equivalent, we store any `e` caught by `} catch( e ) { ` and that could be falsy, e.g.:

```js
try {
  throw null;
} catch(e) {
  console.log(e);
}
```